components/region_cache_memory_engine/src/background.rs

```diff
@@ -549,27 +584,32 @@ impl BackgroundRunnerCore {
             let mut evicted_range = false;
             {
                 let mut engine_wr = self.engine.write();
-                let ranges_to_delete = engine_wr.mut_range_manager().evict_range(range);
-                if !ranges_to_delete.is_empty() {
-                    info!("evict on soft limit reached"; "range" => ?&range, "approx_size" => approx_size, "remaining" => remaining);
+                let mut ranges = engine_wr.mut_range_manager().evict_range(range);
+                evicted_range = !ranges.is_empty();
+                if !ranges.is_empty() {
+                    info!(
+                        "evict on soft limit reached";
+                        "range_to_evict" => ?&range,
+                        "ranges_evicted" => ?ranges,
+                        "approx_size" => approx_size,
+                        "remaining" => remaining
+                    );
                     remaining = remaining
                         .checked_sub(*approx_size as usize)
                         .unwrap_or_default();
-                    // We need to delete the range manually here.
-                    // TODO (afeinberg): consider making delete_range return number of bytes
-                    // deleted.
-                    for r in ranges_to_delete {
-                        engine_wr.engine().delete_range(&r);
-                        if r.overlaps(range) {
-                            evicted_range = true;
-                        }
-                    }
+                    ranges_to_delete.append(&mut ranges);
                 }
             }
             if evicted_range {
```

Comment:
I see the previous code shows that this bool is true when `evict_range` returns true. I am not sure what it means: `evict_range` returninig true means there's no ongoing snapshot so that the range can be deleted immediately. So what exactly does the parameter in `handle_range_evicted` means, is it just be evicted or ready to be deleted? @afeinberg